packages/cdktf-cli/test/get/generator/__snapshots__/complex-computed-types.test.ts.snap

```diff
@@ -159,21 +155,21 @@ export class AcmCertificate extends TerraformResource {
   // tags - computed: false, optional: true, required: false
   private _tags?: { [key: string]: string };
   public get tags() {
-    return this._tags;
+    return this.interpolationForAttribute('tags') as any;
   }
   public set tags(value: { [key: string]: string } | undefined) {
     this._tags = value;
   }
```

Comment:
I'm wondering how I'd add a tag to this resource now:

```ts
const foo = new Resource(this, 'foo', { tags: {abc: "defg"}})
// this was possible before 
foo.tags = Object.assign({}, foo.tags, {bar: 'baz'})
```

Since we just get a reference now - and can't access the supplied value anymore - this would have to be done via terraform functions. 

```ts
// this could probably work
foo.tags = `merge(${foo.tags}, {bar='baz'})`
```

I don't think that's a bad way of doing this in general, but we're lacking the abstractions to make this convenient to use at the moment. That's probably the point where we need intrinsic functions and properly working tokens.

For now, I'd lean towards making the private accessor public - as last resort of accessing the supplied values. The inline documentation could make it clear, that this is a workaround and that there are drawbacks of using this. 

What do you think?

Reply:
I've been thinking about this more from the HCL perspective where editing after the fact isn't a thing, but that does seem like something reasonable to support.

I think making it public works as a short term patch. Fully working functions/tokens would be better, but that's a bigger project.

We'll probably want to do more with documentation in the future, but I'll try to add a note now.

I'll also merge in master to get the new integration test working.